src/main/kotlin/me/partlysanestudios/partlysaneskies/features/discord/webhooks/WebhookMenu.kt

```diff
@@ -0,0 +1,356 @@
+
+            x += width + pad
+            if (x + width > 300) {
+                x = startX
+                y += height + pad
+            }
+        }
+
+        x = startX
+        y = 25.0
+        for (icon in enabled) {
+            icon.iconBox
+                .setX(x.scaledPixels)
+                .setY(y.scaledPixels)
+                .setWidth(width.scaledPixels)
+                .setHeight(height.scaledPixels)
+                .setChildOf(activeWebhooksPanel)
+
+            x += width + pad
+            if (x > 300) {
+                x = startX
+                y += height + pad
+            }
+        }
+
+        for (parameter in activeParameters) {
+            parameter.syncToggle()
+        }
+    }
+
+
+    override fun onDrawScreen(matrixStack: UMatrixStack, mouseX: Int, mouseY: Int, partialTicks: Float) {
+        super.onDrawScreen(matrixStack, mouseX, mouseY, partialTicks)
+
+        for (icon in webhookIcons) {
+            if (icon.hovering && icon.text.isNotEmpty()) {
+                drawHoveringText(icon.text, mouseX, mouseY)
+                break
+            }
+        }
+
+        for (option in activeParameters) {
+            if (option.hovering && option.text.isNotEmpty()) {
+                drawHoveringText(option.text, mouseX, mouseY)
+                break
+            }
+        }
+    }
+
+    private class WebhookIcon(val webhookEvent: Webhook) {
+        var hovering = false
+        var menu: WebhookMenu? = null
+        var text = arrayListOf("§d${webhookEvent.name}")
+        var enabled: Boolean get() {
+            return webhookEvent.enabled
+        } set(value) {
+            webhookEvent.enabled = value
+        }
+        val iconBox = UIBlock().constrain {
+        }.onMouseEnter {
+            hovering = true
+        }.onMouseLeave {
+            hovering = false
+        }.onMouseClick {
+            if (menu?.selectedIcon == this@WebhookIcon) {
+                menu?.selectedIcon = null
+            } else {
+                menu?.selectedIcon = this@WebhookIcon
+            }
+            menu?.updateSelected()
+        }
+
+        val toggle = PSSToggle()
+            .setHeight(100.percent)
+            .setWidth(100.percent)
+            .setX(CenterConstraint())
+            .setY(CenterConstraint())
+            .setChildOf(iconBox)
+
+        init {
+            webhookEvent.icon.constrain {
+                x = CenterConstraint()
+                y = CenterConstraint()
+                height = 100.percent
+                width = 100.percent
+            }
+            webhookEvent.icon.setChildOf(toggle.component)
+        }
+    }
+
+    private class WebhookOption(val config: Config, val optionPath: String) {
+        var hovering = false
+        var menu: WebhookMenu? = null
+
+        val parameterBlock = UIBlock().onMouseEnter {
+            hovering = true
+        }.onMouseLeave {
+            hovering = false
+        }
+
+        val toggle = PSSToggle()
```

Comment:
are you able to use .contraint{} on this?

Reply:
No

Reply:
Okay